extension/src/background/helpers/cachedFetch.ts

```diff
@@ -10,24 +10,28 @@ export const cachedFetch = async (url: string, storageKey: string) => {
   const time = date.getTime();
   const sevenDaysAgo = time - 7 * 24 * 60 * 60 * 1000;
 
-  let directoryLookupJson = JSON.parse(
-    (await dataStorageAccess.getItem(storageKey)) || "{}",
-  );
+  let directoryLookup = (await dataStorageAccess.getItem(storageKey)) || "{}";
+
+  if (typeof directoryLookup === "string") {
+    try {
+      const directoryLookupJSON = JSON.parse(directoryLookup);
+      directoryLookup = directoryLookupJSON;
+    } catch (e) {
+      console.error(`cachedFetch JSON parse error: ${e}`);
```

Comment:
I noticed that `cachedAssetIconsId` was still being stored as a string, I put in this check (similar to one @acharb had proposed) to try to JSON parse if needed to make sure the output here is always an object